File: src/main/java/no/nav/melosys/eessi/models/sed/SED.java

package no.nav.melosys.eessi.models.sed;


import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver;
import lombok.Data;
import no.nav.melosys.eessi.models.sed.medlemskap.Medlemskap;
import no.nav.melosys.eessi.models.sed.nav.Nav;

@SuppressWarnings("unused")
@JsonInclude(Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@Data
public class SED {

    @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "sed")
    @JsonTypeIdResolver(MedlemskapTypeResolver.class)
    private Medlemskap medlemskap;

    private Nav nav;

    private String sed;

    private String sedGVer;

    private String sedVer;
}

File: src/main/java/no/nav/melosys/eessi/models/sed/SED.kt
package no.nav.melosys.eessi.models.sed


import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver


data class SED (
    @field:JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "sed")
    @field:JsonTypeIdResolver(MedlemskapTypeResolver::class)
    var medlemskap: Medlemskap? = null,
    var nav: Nav? = null,
    var sed: String? = null,
    var sedGVer: String? = null,
    var sedVer: String? = null
)